### DISCUSSION TITLE
Support for "Virtual Accounts"

### DISCUSSION THREAD

**Discussion by TheFlagen430297** [Zrips/CMI#7337] [Feature Requests]
# Support for "Virtual Accounts"
Hello! I am running the latest [**CMI**](https://www.spigotmc.org/resources/cmi-298-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/ "The latest version at the time of writing this is 9.2.4.8") and a plugin called [**GriefDefender** (GD)](https://www.spigotmc.org/resources/1-12-2-1-19-2-griefdefender-claim-plugin-grief-prevention-protection.68900/ "A GriefPrevention rewrite that includes a lot more features").
I am also using **CMI** for the economy as well.

Well, when I try to use things such as *selling a plot of land*, **GD** gives this error:
![image](https://user-images.githubusercontent.com/60035052/198279519-1a616852-ff51-4303-a8f2-a9250127dc2c.png)

So, because of that message, I assume I have to ask you guys to add support Virtual Accounts?

## *"This is a current feature in CMI"*
Oh? Okay... How does on enable it?

## *"This cannot be added to CMI"*
I understand if such a system cannot be added to **CMI**.
However, that does leave me with a new question: **What economy plugin should I use?**

**GD** recommends [**EssentialsX**](https://www.spigotmc.org/resources/essentialsx.9089/ "The plugin that is dogwater compared to CMI XD"), however, won't this cause major issues with **CMI**?
Should I go and find random economy plugins until one works? Is there one that will work best with both **CMI** and **GD**? Should I just use **EssentialsX** for economy support??

### In any case thank you so much for just reading this!

**Comment by SickProdigy** [comment]
is a virtual account a bank? Not even sure what that means. EssentialsX is replaced by CMI generally. Have you setup the economy correctly for cmi? There are a couple ways it can be done using vault or cmi alone.

> The Economy engine and Chat manager are disabled by default. You can enable them in the config.yml file. You might need to perform a full server restart for it to take full effect.

https://www.zrips.net/cmi/
This link explains the economy and how to get it fully integrated and working. Read over and make sure you followed it.

Personally, i would use residence with cmi for grief prevention. They are by the same dev. Just my thought though.

Not sure if GD works with vault though. but if it does, should work with CMI also.

**Reply by TheFlagen430297** [reply]
> Have you setup the economy correctly for cmi?
> https://www.zrips.net/cmi/
> This link explains the economy and how to get it fully integrated and working. Read over and make sure you followed it.

I enabled the economy in the config.yml and am using the Modded Vault for **CMI**

> is a virtual account a bank?

Yeah, in **GD** there is the option to make towns, and these towns can have banks, however, they require a "Virtual Account".
I talked to the owner of **GD** and he said that many economy plugins don't support "Virtual Account" but would be easy to add support. *(His words not mine lol)*

> Not sure if GD works with vault though. but if it does, should work with CMI also.

He said it does, and that would be my thought too...

**Comment by mrfloris** [comment]
It would be nice if CMI's economy engine could expand to support 'banks" for things like towny and alike plugins.
